File: Leetcode/Easy/14. Longest Common Prefix/code.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        l=[]
        self.strs=strs
        for i in range(len(strs)):
            l.append(len(strs[i]))        #calculating length of every string in the array
        a=min(l)
        for i in range(a,-1,-1):
            for j in strs:
                if j.startswith(strs[0][:i]):      #checking for common prefix
                    flag=1
                    m=strs[0][:i]       #updating the prefix
                else:
                    flag=0
                    break
            else:
                return m

#Time Complexity: O(N^2)

File: Codeforces/467A. George and Accomodation/code.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    int count=0;
    while (n--){
        int p,q;
        cin>>p>>q;
        if (p<q-1){
            count++;
        }
    }
    cout<<count<<endl;
    return 0;
}

//Time Complexity: O(N)
